csharp/test/Microsoft.ML.OnnxRuntime.Tests/InferenceTest.cs

```diff
@@ -1849,60 +1849,62 @@ private void TestWeightSharingBetweenSessions()
             var dims = new long[] { 3, 2 };
             var dataBuffer = new float[] { 1.0F, 2.0F, 3.0F, 4.0F, 5.0F, 6.0F };
             var allocator = OrtAllocator.DefaultInstance;
-            var ortAllocationInput = allocator.Allocate((uint)dataBuffer.Length * sizeof(float));
-            unsafe
+            using (var ortAllocationInput = allocator.Allocate((uint)dataBuffer.Length * sizeof(float)))
             {
-                float* p = (float*)ortAllocationInput.DangerousGetHandle();
-                for (int i = 0; i < dataBuffer.Length; ++i)
+                unsafe
                 {
-                    *p++ = dataBuffer[i];
+                    float* p = (float*)ortAllocationInput.DangerousGetHandle();
+                    for (int i = 0; i < dataBuffer.Length; ++i)
+                    {
+                        *p++ = dataBuffer[i];
+                    }
                 }
-            }
-            var dataBufferNumBytes = (uint)dataBuffer.Length * sizeof(float);
-            var sharedInitializer = OrtValue.CreateTensorValueWithData(ortCpuMemInfo, Tensors.TensorElementType.Float,
-            dims, ortAllocationInput.DangerousGetHandle(), dataBufferNumBytes);
+                var dataBufferNumBytes = (uint)dataBuffer.Length * sizeof(float);
```

Comment:
We do not need to use OrtAllocation here. Just create an array, pin memory and that would be it

Reply:
Fixed